oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/InMemoryOAuth2AuthorizedClientService.java

```diff
@@ -29,43 +28,59 @@
  * {@link OAuth2AuthorizedClient Authorized Client(s)} in-memory.
  *
  * @author Joe Grandja
+ * @author Vedran Pavic
  * @since 5.0
  * @see OAuth2AuthorizedClientService
  * @see OAuth2AuthorizedClient
  * @see ClientRegistration
  * @see Authentication
  */
 public final class InMemoryOAuth2AuthorizedClientService implements OAuth2AuthorizedClientService {
-	private final Map<String, OAuth2AuthorizedClient> authorizedClients = new ConcurrentHashMap<>();
+	private final Map<OAuth2AuthorizedClientId, OAuth2AuthorizedClient> authorizedClients;
 	private final ClientRegistrationRepository clientRegistrationRepository;
 
 	/**
 	 * Constructs an {@code InMemoryOAuth2AuthorizedClientService} using the provided parameters.
 	 *
+	 * @param authorizedClients the map of authorized clients
 	 * @param clientRegistrationRepository the repository of client registrations
 	 */
-	public InMemoryOAuth2AuthorizedClientService(ClientRegistrationRepository clientRegistrationRepository) {
+	public InMemoryOAuth2AuthorizedClientService(
+			Map<OAuth2AuthorizedClientId, OAuth2AuthorizedClient> authorizedClients,
```

Comment:
Given Map is an optional parameter, I'd prefer it to be exposed via a setter. I understand that means it is mutable object, but this is the typical pattern Spring uses.

Reply:
I agree with @rwinch that we should expose via setter instead - `Map` should be optional. Also, this object is mutable anyway given `saveAuthorizedClient` and `removeAuthorizedClient`.

Reply:
@vpavic Apologies for back-tracking my previous comment but what are your thoughts on removing the setter and adding a 2nd constructor for both `ClientRegistrationRepository` and the `Map`?